topics/getting_started_ktor_client_chat.md

```diff
@@ -102,64 +115,70 @@ fun main() {
 }
 ```
 
-Here, we first create an `HttpClient` and set up Ktor's `WebSocket` plugin (the analog of installing the `WebSocket` plugin in our server application's module in an earlier chapter). Functions in Ktor responsible for making network calls use the suspension mechanism from Kotlin's coroutines, so we wrap our network-related code in a `runBlocking` block. Inside the WebSocket handler, we once again process incoming messages and send outgoing messages: we ignore frames which do not contain text, read incoming text, and send the user input to the server.
+Here, you create an `HttpClient` and install Ktor's `WebSockets` plugin to enable connections to endpoints responding to the WebSocket protocol.
 
-However, this "straightforward" implementation actually contains an issue that prevents it from being used as a proper chat client: when invoking `readLine()`, our program waits until the user enters a message. During this time, we can't see any messages which have been typed out by other users. Likewise, because we invoke `readLine()` after every received message, we would only ever see one new message at a time. Let's address this issue and build a better solution!
+Functions in Ktor responsible for making network calls use the suspension mechanism from Kotlin's coroutines. Therefore, all network-related code is wrapped in a [runBlocking](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/run-blocking.html) block.
+
+Inside the WebSocket handler, the client processes incoming messages by ignoring frames which do not contain text and prints the message. Then, it reads the user's input and sends it to the server.
```

Comment:
> Inside the WebSocket handler, the client processes incoming messages by ignoring frames...

This sounds strange. The client processes incoming messages **by** ignoring frames.

Reply:
yep, it's quite complex. I would try it as `client process only text frames...`

Reply:
Yes, you are right, this is not very clear. In this example, only one frame is handled at a time. Perhaps a better way to say it would be "Inside the WebSocket session handler the client checks if the received `Frame` contains text and prints it out if it does."